components/persistor/lib/mongo/db.ts

```diff
@@ -38,29 +38,37 @@ module.exports = function (PersistObjectTemplate) {
     PersistObjectTemplate.deleteFromMongoQuery = function(template, query, _logger) {
         var db = this.getDB(this.getDBAlias(template.__collection__)).connection;
         var collection = db.collection(this.dealias(template.__collection__));
-        return collection.remove(query, {w:1, fsync:true});
+        return collection.deleteMany(query, {w:1, fsync:true});
     };
 
-    PersistObjectTemplate.getPOJOFromMongoQuery = function(template, query, options, logger) {
+    PersistObjectTemplate.getPOJOFromMongoQuery = async function(template, query, options, logger) {
         (logger || this.logger).debug({component: 'persistor', module: 'db', activity: 'read'}, 'db.' + template.__collection__ + '.find({" + JSON.stringify(query) + "})');
         var db = this.getDB(this.getDBAlias(template.__collection__)).connection;
         var collection = db.collection(this.dealias(template.__collection__));
         options = options || {};
         if (!options.sort)
             options.sort = {_id:1};
-        return collection.find(query, null, options);
+
+        if (typeof(options) === "function") {
+            const res =  await collection.find(query, undefined, options).toArray();
+            return res;
```

Comment:
I know this variables might've been created for debug purpose, but can we directly return the promise

Reply:
Sure!